components/raftstore/src/coprocessor/split_check/mod.rs

```diff
@@ -85,8 +85,47 @@ impl<'a, E> Host<'a, E> {
         Ok(vec![])
     }
 
+    pub fn approximate_bucket_keys<Kv: engine_traits::KvEngine>(
+        &mut self,
+        region: &Region,
+        engine: &Kv,
+    ) -> Result<Vec<Vec<u8>>> {
+        let region_size = if let Ok(size) = get_region_approximate_size(engine, region, 0) {
+            size
+        } else {
+            self.cfg.region_max_size.0
+        };
+
+        const MIN_BUCKET_COUNT_PER_REGION: u64 = 2;
+        const SAMPLE_COUNT_PER_BUCKET: u64 = 10;
+        if region_size >= self.cfg.region_bucket_size.0 * MIN_BUCKET_COUNT_PER_REGION {
+            let mut bucket_checker = size::Checker::new(
+                self.cfg.region_bucket_size.0 * MIN_BUCKET_COUNT_PER_REGION
+                    / SAMPLE_COUNT_PER_BUCKET,
+                self.cfg.region_bucket_size.0 / SAMPLE_COUNT_PER_BUCKET,
```

Comment:
useless

Reply:
which part is useless here?
Do you mean SAMPLE_COUNT_PER_BUCKET? This is to mitigate the issue that that split keys are not appropriate when the SST files have overlapped ranges.

Reply:
`Checker::approximate_split_keys()` doesn't use `max_size` and `split_size`. It calls `get_approximate_split_keys()` with `batch_split_limit` directly.